cypress/e2e/seededFlows/articleFlows/postModerationTools.spec.js

```diff
@@ -366,5 +366,81 @@ describe('Moderation Tools for Posts', () => {
         );
       });
     });
+
+    describe('flag-user flow', () => {
+      beforeEach(() => {
+        cy.get('@trustedUser').then((user) => {
+          cy.loginAndVisit(user, '/admin_mcadmin/test-article-slug');
+          cy.findByRole('heading', { level: 1, name: 'Test article' });
+          cy.findByRole('button', { name: 'Moderation' }).click();
+        });
+      });
+
+      it('should show error message if flag-user radio is unchecked', () => {
+        cy.getIframeBody('[title="Moderation panel actions"]').within(() => {
+          cy.findByRole('button', { name: 'Open admin actions' })
+            .as('moderatingActionsButton')
+            .pipe(click)
+            .should('have.attr', 'aria-expanded', 'true');
+          cy.findByRole('button', { name: 'Flag admin_mcadmin' }).click();
+        });
+        cy.findByRole('dialog').within(() => {
+          cy.findByRole('button', { name: 'Confirm Flag' }).click();
+          cy.findByTestId('unselected-radio-error')
+            .contains('You must check the radio button first.')
+            .should('exist');
+        });
+      });
+
+      it('should flag the user if flag-user radio is checked', () => {
+        cy.getIframeBody('[title="Moderation panel actions"]').within(() => {
+          cy.findByRole('button', { name: 'Open admin actions' })
+            .as('moderatingActionsButton')
+            .pipe(click)
+            .should('have.attr', 'aria-expanded', 'true');
+          cy.findByRole('button', { name: 'Flag admin_mcadmin' }).click();
+        });
+
+        cy.findByRole('dialog').within(() => {
+          cy.get('#flag-user-radio-input').check();
```

Comment:
@aitchiss I tried adding a `data-testid` to the radio button in the ERB file, but Cypress kept complaining it couldn't find it. 😓  When I used a `get` with the `id` it worked, but I KNOW this is not best practice.

Reply:
As a general rule, first port of call should be `findByRole` - I don't know why `data-testid` was having issues (I suspect maybe there was another element with the same test id? IDK), but you can target by the role -

```suggestion
         cy.findByRole('radio', {
            name: "Make all posts by admin_mcadmin less visible admin_mcadmin consistently posts content that violates DEV(local)'s code of conduct because it is harassing, offensive or spammy.",
          }).click();
```

The name is _very_ long, and probably should just be "Make all posts by admin_mcadmin less visible" but it seems like making changes to that accessible name isn't really in scope, and this is fine enough as it is